### PR TITLE
/cluster: cleanup clientbin.sh

### PR DESCRIPTION
**Pull request by neolit123**
**What this PR does / why we need it**:

/kind cleanup
/priority backlog

clientbin.sh is only used in kubectl.sh. It contains logic
for finding a binary that was build from the k/k tree
and for printing errors if the binary cannot be found.

- Remove clientbin.sh
- Remove references of clientbin.sh from /cluster/BUILD
- Use hack/lib/util.sh's kube::util::find-binary for finding the binary
(NOTE: /cluster already sources /hack)

**Which issue(s) this PR fixes**:

NONE

**Special notes for your reviewer**:
NONE

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: cluster/kubectl.sh [hunk 1]
```diff
@@ -32,15 +32,17 @@ set -o pipefail
...
 KUBE_ROOT=${KUBE_ROOT:-$(dirname "${BASH_SOURCE[0]}")/..}
 source "${KUBE_ROOT}/cluster/kube-util.sh"
-source "${KUBE_ROOT}/cluster/clientbin.sh"
+source "${KUBE_ROOT}/hack/lib/util.sh"
 
 # If KUBECTL_PATH isn't set, gather up the list of likely places and use ls
 # to find the latest one.
 if [[ -z "${KUBECTL_PATH:-}" ]]; then
-  kubectl=$( get_bin "kubectl" "cmd/kubectl" )
+  kubectl=$( kube::util::find-binary "kubectl" )
```
Comment (BenTheElder) [inline cluster/kubectl.sh:40]: did you check that these have the same search paths etc?

File: cluster/kubectl.sh [hunk 2]
```diff
@@ -32,15 +32,17 @@ set -o pipefail
...
 
   if [[ ! -x "$kubectl" ]]; then
-    print_error "kubectl"
+    {
+      echo "It looks as if you don't have a compiled kubectl binary"
```
Comment (BenTheElder) [inline cluster/kubectl.sh:44]: maybe inline the previous error message to match?

File: cluster/kubectl.sh (hunk 1 above)
Comment (neolit123) [reply cluster/kubectl.sh:40]: the logic looks the same.
`::find-binary`:
https://github.com/kubernetes/kubernetes/blob/master/hack/lib/util.sh#L147-L221
`clientbin.sh`:
https://github.com/kubernetes/kubernetes/blob/master/cluster/clientbin.sh#L23-L96

File: cluster/kubectl.sh (hunk 2 above)
Comment (neolit123) [reply cluster/kubectl.sh:44]: this the old message:
https://github.com/kubernetes/kubernetes/blob/master/cluster/clientbin.sh#L98-L109

should the whole message be added?

File: cluster/kubectl.sh (hunk 1 above)
Comment (BenTheElder) [reply cluster/kubectl.sh:40]: ack

File: cluster/kubectl.sh (hunk 2 above)
Comment (BenTheElder) [reply cluster/kubectl.sh:44]: I feel like we should continue printing the whole / same message

Comment (neolit123) [reply cluster/kubectl.sh:44]: updated.

**Comment by BenTheElder** [comment]
cc @amwat @cheftako 
/lgtm
/approve

**Comment by BenTheElder** [comment]
thank you!

**Pull request merged**